src/SortingTable/SortingTable.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from "react";
+import { Table, TableProps } from "../Table";
+
+type SortingTableProps = TableProps & {
+  initialSortColumn: string;
+};
+
+const numericAlphabeticalSort = (a, b, numeric) =>
+  String(a).localeCompare(b, undefined, { numeric, sensitivity: "base" });
+
+const applySort = (rows, sortColumn, columns) =>
+  rows.sort((a, b) => {
+    const column = columns.find((col) => col.dataKey === sortColumn);
+    const { numeric } = column;
+
+    return numericAlphabeticalSort(a[sortColumn], b[sortColumn], numeric);
+  });
+
+const sortRows = (rows, columns, sortState) => {
```

Comment:
Are the argument and return types for these functions easy to define? Locally my TypeScript compiler is unable to infer any of them except for the `numericAlphabeticalSort` returning a number.

Reply:
I'm not familiar with typescript, but I have done prop types in that past. Going off of that, the arguments might be difficult since they aren't always the same.
Did you want to pair on it?